Added Support for TitleScreen/Splash Images+Logo+Senario. Fixed Shop Free Pulls.

SideswipeeZ opened the pull request:
Bugs:
- Changed Shop File to Remove 'Free Pull' from two Packs that was causing the game to error.

Adds Support:
- Splashscreen Logos (E.g. 25th Anniversary).
- Images for Card art (on the right-hand side of the client) randomly chosen from the list returned by the server.
- Added scenarios to play if the client is left idle at the splash screen for long enough.

![image](https://github.com/pixeltris/YgoMaster/assets/20016930/0d367721-262c-419f-9f22-b2650e850a29)
![image](https://github.com/pixeltris/YgoMaster/assets/20016930/73ea544b-5ed0-496a-ac61-a39c6aea8641)

pixeltris commented:
The Shop.json entries are generated based on `YgoMaster --mergeshops` which merges old logs I have to prevent having to refetch secret packs (though they are now outdated and I should update them). As I don't manually edit the file I've added a hacky check for these 2 secret packs which I never pulled to fix the issue described here (they were alt art secret packs so nothing should be lost).

The `TitleLoop` sounds like a good addition. It might be better to put in a json file like `YgoMaster/Data/TitleLoop.json` and generate it from the fiddler inspector to keep in sync with the official game. e.g. https://github.com/pixeltris/YgoMaster/blob/b0eb64760a63237e9ae835c2aaba247072900fbe/YgoMasterFiddler/YgoMasterInspector.cs#L437-L440

Merik2013 commented:
very nice. you seem to know your way around the source code much better than I. Can I make a request? 

Yami and I are revamping the LotD story mode and would like to be able to add in custom music without having to unpack and repack localdata/ overwrite preexisting music. We can add in the BGM entries for the soloduel files and I understand that if this feature were implemented into the mod I would place the wav files into the \ClientData\sound\audioclip\bgm folder path. I also understand that they need to be formatted specifically (Load type: Streaming, Compression format: Vorbis, Sample Rate Setting: Override Sample Rate, Sample Rate: 48000Hz). The only issue seems to be that the mod isn't coded to read user provided BGM files like this.

SideswipeeZ commented:
> The Shop.json entries are generated based on `YgoMaster --mergeshops` which merges old logs I have to prevent having to refetch secret packs (though they are now outdated and I should update them). As I don't manually edit the file I've added a hacky check for these 2 secret packs which I never pulled to fix the issue described here (they were alt art secret packs so nothing should be lost).

Yeah, i moved them out manually as i have my server instance with all packs revealed.

> 
> The `TitleLoop` sounds like a good addition. It might be better to put in a json file like `YgoMaster/Data/TitleLoop.json` and generate it from the fiddler inspector to keep in sync with the official game. e.g.
> 
> https://github.com/pixeltris/YgoMaster/blob/b0eb64760a63237e9ae835c2aaba247072900fbe/YgoMasterFiddler/YgoMasterInspector.cs#L437-L440

That is a good idea. I took the fiddler snap before 1.8.0 came out, ill check if its static list or not aswell. Though Im assuming it will update when new cards and arts/senarios are added so ill take a look at adding that to the inspector.

SideswipeeZ commented:
> very nice. you seem to know your way around the source code much better than I. Can I make a request?
> 
> Yami and I are revamping the LotD story mode and would like to be able to add in custom music without having to unpack and repack localdata/ overwrite preexisting music. We can add in the BGM entries for the soloduel files and I understand that if this feature were implemented into the mod I would place the wav files into the \ClientData\sound\audioclip\bgm folder path. I also understand that they need to be formatted specifically (Load type: Streaming, Compression format: Vorbis, Sample Rate Setting: Override Sample Rate, Sample Rate: 48000Hz). The only issue seems to be that the mod isn't coded to read user provided BGM files like this.

This might be out of my scope here but ill take a look and see if I can, but I'm not going to promise anything at this point.

pixeltris commented:
I opened a feature request for the custom audio files #311
